### PR TITLE
Add support for http tokens in metadata endpoint

### PR DESCRIPTION
**Pull request by xorxsan**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -68,3 +68,8 @@ variable "encrypt_ebs" {
   type    = bool
   default = true
 }
+
+variable "require_http_tokens" {
+  type    = bool
+  default = true
```
Comment (betabandido) [inline variables.tf:74]: Why not `false`? Setting it to true breaks backward compatibility.

File: examples/basic/cluster.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  cluster_name       = "simple-eks-integration-test-for-eks-node-group"
+  cluster_version    = "1.20"
+  vpc_name           = var.vpc_name
+  log_group_name     = "a-test-log-group-name"
+
+  region  = var.aws_region
+  profile = var.aws_profile
+}
+
+module "spot_node_group" {
```
Comment (betabandido) [inline examples/basic/cluster.tf:14]: I would do the example with an on-demand node (like t3a.small/medium or something like that). It simplifies the test and also reduces the likelihood of the test failing if no spot instance can be found.

File: test/basic_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+			"vpc_name":    vpcName,
+		},
+		NoColor: true,
+	})
+
+	defer terraform.Destroy(t, terraformOptions)
+
+	terraform.InitAndApply(t, terraformOptions)
+
+	assert.Regexp(t,
```
Comment (betabandido) [inline test/basic_test.go:38]: All of these asserts check values for resources created by the cluster. We should do assertions on resources created by the nodes.

File: examples/basic/cluster.tf [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  cluster_name    = "simple-eks-integration-test-for-eks-node-group"
+  cluster_version = "1.20"
+  vpc_name        = var.vpc_name
+  log_group_name  = "a-test-log-group-name"
+
+  region  = var.aws_region
+  profile = var.aws_profile
+}
+
+module "on_deman_node_group" {
```
Comment (betabandido) [inline examples/basic/cluster.tf:14]: deman => demand

File: outputs.tf [hunk 5]
```diff
@@ -0,0 +1,7 @@
+output "id" {
```
Comment (betabandido) [inline outputs.tf:1]: Do we need these outputs?

Comment (xorxsan) [reply outputs.tf:1]: If we don't add them, then I cannot use them on examples/outputs.tf.

**Pull request merged**